napari/_qt/qt_event_loop.py

```diff
@@ -8,31 +8,64 @@
 
 from ..utils.perf import perf_config
 from .exceptions import ExceptionHandler
+from .qthreading import wait_for_workers_to_quit
 
 
-def _create_application(argv) -> QApplication:
-    """Create our QApplication.
+def get_app() -> QApplication:
+    """Get or create the Qt QApplication
 
     Notes
     -----
     Substitute QApplicationWithTracing when using perfmon.
 
     With IPython/Jupyter we call convert_app_for_tracing() which deletes
     the QApplication and creates a new one. However here with gui_qt we
-    need to create the correct QApplication up front, or we will crash.
-    We'll crash because we'd be deleting the QApplication after we created
-    QWidgets with it, such as we do for the splash screen.
+    need to create the correct QApplication up front, or we will crash because
+    we'd be deleting the QApplication after we created QWidgets with it,
+    such as we do for the splash screen.
     """
-    if perf_config and perf_config.trace_qt_events:
-        from .perf.qt_event_tracing import QApplicationWithTracing
+    app = QApplication.instance()
+    if app:
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import convert_app_for_tracing
 
-        return QApplicationWithTracing(argv)
+            # no-op if app is already a QApplicationWithTracing
+            app = convert_app_for_tracing(app)
+        app._existed = True
     else:
-        return QApplication(argv)
+        # automatically determine monitor DPI.
+        # Note: this MUST be set before the QApplication is instantiated
+        QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)
+
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import QApplicationWithTracing
+
+            app = QApplicationWithTracing(sys.argv)
+        else:
+            app = QApplication(sys.argv)
+
+        # if this is the first time the Qt app is being instantiated, we set
+        # the name, so that we know whether to raise_ in Window.show()
+        from napari import __version__
+
+        app.setApplicationName('napari')
+        app.setApplicationVersion(__version__)
+        app.setOrganizationName('napari')
+        app.setOrganizationDomain('napari.org')
+
+    if perf_config and not perf_config.patched:
+        # Will patch based on config file.
+        perf_config.patch_callables()
+
+    # see docstring of `wait_for_workers_to_quit` for caveats on killing
+    # workers at shutdown.
+    app.aboutToQuit.connect(wait_for_workers_to_quit)
+
+    return app
 
 
 @contextmanager
-def gui_qt(*, startup_logo=False, gui_exceptions=False):
+def gui_qt(*, startup_logo=False, gui_exceptions=False, force=False):
```

Comment:
If people aren't using `gui_qt` anymore is there a way for them to set `startup_logo` and `gui_exceptions`? Should these be passed to `run_app`? I think probably not - I'd probably rather have them settable in our global settings once #1183 goes in. For now we could add them as env variables `NAPARI_STARTUP_LOGO` and `NAPARI_GUI_EXCEPTIONS`?

Reply:
I think startup logo probably only belongs in `__main__`, and if it's there, then I don't think it needs an option at all, since it will only show (appropriately) when you're starting "the napari application" from command line or from bundle, etc...
gui exceptions feels like something better suited for `napari.config`, and an env var would be ok there too (though I wonder if maybe we're getting a little trigger happy with environment variables as configurators?)

Reply:
both of those approaches make sense to me. Not sure if they make sense to address here, in other PRs etc. I'd say fairly low priority, but I don't want them to get lost